.github/workflows/event-merge-to-queue.yml

```diff
@@ -7,7 +7,9 @@ on:
     types: [checks_requested]
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.event.merge_group.base_ref }}
+  # This group identifies the PR by the tree ID of the head commit.
+  # TODO: Replace with PR number of branch name once GH adds it to the event context
+  group: ${{ github.workflow }}-${{ github.event.merge_group.head_commit.tree_id }}
```

Comment:
The head commit of the tree id should identify the same PR in different takes (after it goes in and out of the merge queue) right? How does it work exactly? Should this go in the other branches as well?

Reply:
It already moved. This works partially, but it’s the best we can currently do, since no better option is available by GitHub ATM. It’s a requested feature, and I also saw some actions that attempt to do it, maybe it will work better